### PR TITLE
CASSANDRA-16361 DROP COMPACT STORAGE should invalidate prepared statements still using CompactTableMetadata

### PR DESCRIPTION
**Pull request by maedhroz**

### CODE REVIEW

File: src/java/org/apache/cassandra/schema/TableMetadata.java [hunk 1]
```diff
@@ -558,7 +558,8 @@ boolean changeAffectsPreparedStatements(TableMetadata updated)
             || !regularAndStaticColumns.equals(updated.regularAndStaticColumns)
             || !indexes.equals(updated.indexes)
             || params.defaultTimeToLive != updated.params.defaultTimeToLive
-            || params.gcGraceSeconds != updated.params.gcGraceSeconds;
+            || params.gcGraceSeconds != updated.params.gcGraceSeconds
+            || flags.contains(Flag.DENSE) && !updated.flags.contains(Flag.DENSE);
```
Comment (ifesdjeen) [inline src/java/org/apache/cassandra/schema/TableMetadata.java]: You could use `!Flag.isCQLTable(flags)` instead:  https://github.com/apache/cassandra/blob/61046b5c6836d86f38d238e3629074099a6a6545/src/java/org/apache/cassandra/schema/TableMetadata.java#L99

Comment (michaelsembwever) [reply src/java/org/apache/cassandra/schema/TableMetadata.java]: can we also put it in parenthesis for readability…

```suggestion
            || ( !isCQLTable(flags) && isCQLTable(updated.flags) );
```

File: src/java/org/apache/cassandra/cql3/QueryProcessor.java [hunk 2]
```diff
@@ -161,6 +161,12 @@ public static void clearPreparedStatements(boolean memoryOnly)
         if (!memoryOnly)
             SystemKeyspace.resetPreparedStatements();
     }
+
```
Comment (michaelsembwever) [inline src/java/org/apache/cassandra/cql3/QueryProcessor.java:164]: fyi: white-space here i had to remove. there was also three lines with trailing whitespace introduced in CompactTableTest, also fixed.

**Comment by maedhroz** [comment]
Committed as https://github.com/apache/cassandra/commit/d9859d231f24781b7ebf4dd93bb4584fb4bc7538

**Pull request closed without merging**